src/Orleans/Threading/CancellationTokenHolderGrain.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Placement;
+using Orleans.Runtime;
+
+namespace Orleans.Threading
+{
+    [PreferLocalPlacement]
+    internal class CancellationTokenHolderGrain : Grain, ICancellationTokenHolderGrain
+    {
+        private CancellationTokenSource _cancellationTokenSource;
+        private bool _collectionScheduled;
+        private TimeSpan _safetyCautionLimit = TimeSpan.FromSeconds(1);
+
+        public override Task OnActivateAsync()
+        {
+            DelayDeactivation(TimeSpan.FromDays(5));
```

Comment:
I've seen tasks that were running for more than 20 hours. 5 days is just to be on the safe side..